common/math/math.go

```diff
@@ -34,3 +36,165 @@ func RoundFloat(x float64, prec int) float64 {
+	var diffs []float64
+	for i := range values {
+		diffs = append(diffs, values[i]-benchmarkRates[i])
+	}
+	stdDev := PopulationStandardDeviation(diffs)
+	if stdDev == 0 {
+		return 0
+	}
+	return (averageValues - averageComparison) / stdDev
+}
+
+// PopulationStandardDeviation calculates standard deviation using population based calculation
+func PopulationStandardDeviation(values []float64) float64 {
+	if len(values) == 0 {
+		return 0
+	}
+	avg := ArithmeticAverage(values)
+	diffs := make([]float64, len(values))
+	for x := range values {
+		diffs[x] = math.Pow(values[x]-avg, 2)
+	}
+	return math.Sqrt(ArithmeticAverage(diffs))
+}
+
+// SampleStandardDeviation standard deviation is a statistic that
+// measures the dispersion of a dataset relative to its mean and
+// is calculated as the square root of the variance
+func SampleStandardDeviation(vals []float64) float64 {
+	if len(vals) <= 1 {
+		return 0
+	}
+	mean := ArithmeticAverage(vals)
+	var superMean []float64
+	var combined float64
+	for i := range vals {
+		result := math.Pow(vals[i]-mean, 2)
+		superMean = append(superMean, result)
+		combined += result
+	}
+	avg := combined / (float64(len(superMean)) - 1)
+	return math.Sqrt(avg)
+}
+
+// GeometricAverage is an average which indicates the central tendency or
+// typical value of a set of numbers by using the product of their values
+// The geometric average can only process positive numbers
+func GeometricAverage(values []float64) float64 {
+	if len(values) == 0 {
+		return 0
+	}
+	product := 1.0
+	for i := range values {
+		if values[i] <= 0 {
+			// cannot use negative or zero values in geometric calculation
+			return 0
+		}
+		product *= values[i]
+	}
+	geometricPower := math.Pow(product, 1/float64(len(values)))
+	return geometricPower
+}
+
+// FinancialGeometricAverage is a modified geometric average to assess
+// the negative returns of investments. It accepts It adds +1 to each
+// This does impact the final figures as it is modifying values
+// It is still ultimately calculating a geometric average
+// which should only be compared to other financial geometric averages
+func FinancialGeometricAverage(values []float64) float64 {
+	if len(values) == 0 {
+		return 0
+	}
+	product := 1.0
+	for i := range values {
+		if values[i] <= -1 {
+			// cannot lose more than 100%, figures are incorrect
+			return 0
+		}
+		// as we cannot have negative or zero value geometric numbers
+		// adding a 1 to the percentage movements allows for differentiation between
+		// negative numbers (eg -0.1 translates to 0.9) and positive numbers (eg 0.1 becomes 1.1)
+		modVal := values[i] + 1
+		product *= modVal
+	}
+	geometricPower := math.Pow(product, 1/float64(len(values)))
+	// we minus 1 because we manipulated the values to be non-zero/negative
+	return geometricPower - 1
+}
+
+// ArithmeticAverage is the basic form of calculating an average.
+// Divide the sum of all values by the length of values
+func ArithmeticAverage(values []float64) float64 {
```

Comment:
These functions should return an error maybe? Currently there is no way to determine between an actual 0 average and 0 returned because of no input values